third_party/terraform/resources/resource_compute_instance_template.go

```diff
@@ -888,3 +865,28 @@ func resourceComputeInstanceTemplateDelete(d *schema.ResourceData, meta interfac
 	d.SetId("")
 	return nil
 }
+
+func expandResourceComputeInstanceTemplateScheduling(d *schema.ResourceData, meta interface{}) (*computeBeta.Scheduling, error) {
+	v, ok := d.GetOk("scheduling")
+	if !ok || v == nil {
+		return &computeBeta.Scheduling{
+			OnHostMaintenance: "MIGRATE",
+		}, nil
+	}
+
+	schedulings := v.([]interface{})
+	if len(schedulings) > 1 {
+		return nil, fmt.Errorf("Unable to expand scheduling, more than one `scheduling` block is defined")
+	}
+
+	expanded, err := expandScheduling(v)
+	if err != nil {
+		return nil, err
+	}
+
+	if expanded.Preemptible && expanded.OnHostMaintenance == "" {
+		expanded.OnHostMaintenance = "TERMINATE"
+		expanded.ForceSendFields = append(expanded.ForceSendFields, "OnHostMaintenance")
```

Comment:
I'm curious about why this ForceSendFields is necessary given that we just set OnHostMaintenance to a value on the previous line

Reply:
tbh i didn't test it without it - most of this was refactored while trying to not change the behavior of the previous code. I can remove it since I agree that it isn't doing anything.